trace/HTTP.md

```diff
@@ -0,0 +1,114 @@
+
+## Message events
+
+In the lifetime of an incoming and outgoing request, the following
+message events SHOULD be created:
+
+* A message event for the request body size if/when determined.
+* A message event for the response size if/when determined.
+
+Implementations SHOULD create message event when body size is determined.
+
+```
+-> [time], MessageEventTypeSent, UncompressedByteSize, CompressedByteSize
+```
+
+Implementations SHOULD create message event when response size is determined.
+
+```
+-> [time], MessageEventTypeRecv, UncompressedByteSize, CompressedByteSize
+```
+
+## Attributes
+
+Implementations SHOULD set the following attributes on the client
+and server spans. For a server, request represents the incoming request.
+For a client, request represents the outgoing request.
+
+All attributes are optional.
+
+| Attribute name            | Description                 | Example value                   |
+|---------------------------|-----------------------------|---------------------------------|
+| "http.host"               | Request URL host            | "example.com"                   |
+| "http.port"               | Request URL port            | 443                             |
+| "http.method"             | Request URL method          | "GET"                           |
+| "http.path"               | Request URL path            | "/users/25f4c31d"               |
+| "http.route"              | Matched request URL route   | "/users/:userID"                |
+| "http.user_agent"         | Request user-agent          | "HTTPClient/1.2"                |
+| "http.status_code"        | Response status code        | 200                             |
+
+Exporters should always export the collected attributes.
+Exporters should map the collected attributes to backend's
+known attributes/labels.
+
+The following table summarizes how OpenCensus attributes maps to the
+known attributes/labels on supported tracing backends.
+
+| OpenCensus attribute      | Zipkin             | Jaeger             | Stackdriver Trace label   |
+|---------------------------|--------------------|--------------------|---------------------------|
+| "http.host"               | "http.host"        | *                  | "/http/host"              |
+| "http.method"             | "http.method"      | "http.method"      | "/http/method"            |
+| "http.path"               | "http.path"        | *                  | "/http/path"              |
+| "http.route"              | "http.route"       |                    | "/http/route"             |
+| "http.user_agent"         |                    |                    | "/http/user_agent"        |
+| "http.status_code"        | "http.status_code" | "http.status_code" | "/http/status_code"       |
+
+
+(*) Jager supports "http.url" instead of a path and method attribute.
+OpenCensus doesn't collect the full URL for security reasons.
+Exporters can build a URL from the collected method and path and
```

Comment:
this will work for jaeger assuming they've defined it w/o query params and you can infer the scheme. This won't work for zipkin